db/dcrpg/queries.go

```diff
@@ -436,35 +490,47 @@ func InsertVotes(db *sql.DB, dbTxns []*dbtypes.Tx, _ /*txDbIDs*/ []uint64, fTx *
 
 		var rowID uint64
 		for _, val := range choices {
-			index, err := dbtypes.ChoiceIndexFromStr(val.Choice.Id)
-			if err != nil {
-				bail()
-				return nil, nil, nil, nil, nil, err
-			}
+			// Update the Status if its not upto date.
+			p := votesMilestones.AgendaMileStones[val.ID]
+			s := storedAgendas[val.ID]
+			if s.Status != p.Status {
+				err = agendaStmt.QueryRow(val.ID, p.Status,
+					p.VotingDone, p.Activated, p.HardForked).Scan(&s.ID)
+				if err != nil {
+					bail()
+					return nil, nil, nil, nil, nil, fmt.Errorf("agenda INSERT failed: : %v", err)
+				}
 
-			lockedIn, activated, hardForked := false, false, false
+				s.Status = p.Status
+				s.VotingDone = p.VotingDone
+				s.Activated = p.Activated
+				s.HardForked = p.HardForked
+				storedAgendas[val.ID] = s
+			}
 
-			// THIS IS A TEMPORARY SOLUTION till activated, lockedIn and hardforked
-			// height values can be sent via an rpc method.
-			progress, ok := VotingMilestones[val.ID]
-			if ok {
-				lockedIn = (progress.LockedIn == tx.BlockHeight)
-				activated = (progress.Activated == tx.BlockHeight)
-				hardForked = (progress.HardForked == tx.BlockHeight)
+			if p.ID == 0 {
+				p.ID = s.ID
```

Comment:
From what I can tell, this is the only update to `p`.  But it's not clear why InsertVotes is modifying it or generally what is going on with `votesMilestones` or `storedAgendas`.

Reply:
I was looking for a simple way to have the data stored in agendas table i.e `lockedIn`, `Hardforked` and `Activated` heights as up to date as possible without making too many db accesses everytime we are updating the agenda_votes table.
`votesMilestones` holds the updated blockchain info always while `storedAgendas` holds a mirror copy of what in the agendas db. If there is a difference between the two while blockchain sync is in progress, an update is made to agendas table. The update is made only when needed i.e. where there is a status change.

Reply:
"without making too many db accesses everytime we are updating the agenda_votes table" <- that clarifies it.  Please comment code gratuitously to explain what's going on.